mql.go

```diff
@@ -0,0 +1,103 @@
+package mql
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+)
+
+// WhereClause contains a SQL where clause condition and its arguments.
+type WhereClause struct {
+	// Condition is the where clause condition
+	Condition string
+	// Args for the where clause condition
+	Args []any
+}
+
+// Parse will parse the query and use the provide database model to create a
+// where clause. Supported options: WithColumnMap, WithIgnoreFields
+func Parse(query string, model any, opt ...Option) (*WhereClause, error) {
+	const op = "mql.Parse"
+	switch {
+	case query == "":
+		return nil, fmt.Errorf("%s: missing query: %w", op, ErrInvalidParameter)
+	case isNil(model):
+		return nil, fmt.Errorf("%s: missing model: %w", op, ErrInvalidParameter)
+	}
+	p := newParser(query)
+	expr, err := p.parse()
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", op, err)
+	}
+	fValidators, err := fieldValidators(reflect.ValueOf(model), opt...)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", op, err)
+	}
+	e, err := exprToWhereClause(expr, fValidators, opt...)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", op, err)
+	}
+	return e, nil
+}
+
+// exprToWhereClause generates the where clause condition along with it's
+// required arguments. Supported options: WithColumnMap, WithConverter
+func exprToWhereClause(e expr, fValidators map[string]validator, opt ...Option) (*WhereClause, error) {
+	const op = "mql.exprToWhereClause"
+	switch {
+	case isNil(e):
+		return nil, fmt.Errorf("%s: missing expression: %w", op, ErrInvalidParameter)
+	case isNil(fValidators):
+		return nil, fmt.Errorf("%s: missing validators: %w", op, ErrInvalidParameter)
+	}
+
+	switch v := e.(type) {
+	case *comparisonExpr:
+		opts, err := getOpts(opt...)
+		if err != nil {
+			return nil, fmt.Errorf("%s: %w", op, err)
+		}
```

Comment:
Doesn't it make more sense to do this before the type switch?

Reply:
I don't know.  We only use it this particular case at the moment. We can always
move it in the future if it's needed for other bits in the func.